Update access graph version and add Docker deployment warning (#42002)
gravitational/teleport: docs/pages/access-controls/access-graph/self-hosted.mdx

- A TLS certificate for the Access Graph service
  - The TLS certificate must be issued for "server authentication" key usage,
    and must list the IP or DNS name of the TAG service in an X.509 v3 `subjectAltName` extension.

>   - Starting from version 1.20.4 of the Access Graph service, the container by default runs as a non-root user.

Reviewer: nit: "the container runs as a non-root user by default." might be a nicer phrasing?

    Make sure the certificate files are readable by the user running the container. You can set correct permissions with the following command:

>     ```console

Reviewer 2: ```suggestion
    ```code
```
For consistency with the rest of the docs

>     $ sudo chown 65532 /path/to/tls.key

Reviewer: Nit: we use `/etc/access_graph/tls.key` as the path in the example config further below, so maybe use that here as well.

Author: Changed

    ```
- The node running the Access Graph service must be reachable from Teleport Auth Service and Proxy Service.

<Notice type="warning">
    The deployment with Docker is suitable for testing and development purposes. For production deployments,

>     consider using a container orchestration system like Kubernetes and Helm.
>     Refer to [Helm chart for Access Graph](self-hosted-helm.mdx) for instructions.

Reviewer 2: ```suggestion
    consider using the Teleport Access Graph Helm chart to deploy this service on Kubernetes.
    Refer to [Helm chart for Access Graph](self-hosted-helm.mdx) for instructions.
```

Suggested to be more direct about the fact that we maintain a Helm chart for deploying TAG.

```code
$ sudo mkdir /etc/access_graph
$ curl -s 'https://<Var name="teleport.example.com" />/webapi/auth/export?type=tls-host' | sudo tee /etc/access_graph/teleport_host_ca.pem

>     ```

Reviewer: unintended change?
